src/python/pants/util/frozendict_test.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from collections import OrderedDict
+from dataclasses import dataclass
+
+import pytest
+
+from pants.util.frozendict import FrozenDict
+
+
+def test_flexible_constructor() -> None:
+    expected = FrozenDict({"a": 0, "b": 1})
```

Comment:
You copy in the FrozenDict constructor, which is important but untested. It would be good to add a test that constructs with a dict, mutates that dict, then confirms the constructed FrozenDict is not likewise mutated.

Reply:
Added.